### PR TITLE
Allow false/true to be read as keys in Hash::get().

### PR DESCRIPTION
**Pull request by markstory**
While these are not values within the documented types, there exist use cases in CakeSession that necessitate these to be supported types.

Refs #10196

### CODE REVIEW

**Comment by markstory** [comment]
I'm not sure we want to port this behavior to 3.x as it will impede our ability to add stricter types in the future.

**Comment by bancer** [comment]
@markstory A tip: https://help.github.com/articles/closing-issues-via-commit-messages/

**Comment by dereuromark** [comment]
@bancer We know :) Our process still is to close the duplicates here earlier so far.

**Comment by markstory** [comment]
@bancer I'm well aware of how to close issues with commit messages. I prefer to close issues when pull requests are opened, as pull requests create _more_ issues leaving us with double issues.

**Comment by markstory** [comment]
> If the commit is in a non-default branch, the issue will remain open and the issue will be referenced with a tooltip.

This is another reason we close issues when pull requests are opened :smile:

File: lib/Cake/Test/Case/Utility/HashTest.php [hunk 1]
```diff
@@ -249,7 +252,7 @@ public function testGetEmptyKey() {
  * @return void
  */
 	public function testGetInvalidPath() {
-		Hash::get(array('one' => 'two'), true);
+		Hash::get(array('one' => 'two'), new StdClass());
```
Comment (dereuromark) [inline lib/Cake/Test/Case/Utility/HashTest.php]: technically: stdClass (lowercase s)

**Pull request merged**